pkg/cli/cmd/credential/register/aws/accesskey/accesskey.go

```diff
@@ -0,0 +1,166 @@
+	_ = cmd.MarkFlagRequired("secret-access-key")
+
+	return cmd, runner
+}
+
+// Runner is the runner implementation for the `rad credential register aws` command.
+type Runner struct {
+	ConfigHolder      *framework.ConfigHolder
+	ConnectionFactory connections.Factory
+	Output            output.Interface
+	Format            string
+	Workspace         *workspaces.Workspace
+
+	AccessKeyID     string
+	SecretAccessKey string
+	KubeContext     string
+}
+
+// NewRunner creates a new instance of the `rad credential register aws` runner.
+func NewRunner(factory framework.Factory) *Runner {
+	return &Runner{
+		ConfigHolder:      factory.GetConfigHolder(),
+		ConnectionFactory: factory.GetConnectionFactory(),
+		Output:            factory.GetOutput(),
+	}
+}
+
+// Validate runs validation for the `rad credential register aws` command.
+//
+
+// Validate() checks if the required workspace, output format, access key ID and secret access key are present, and if not, returns an error.
+func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
+	workspace, err := cli.RequireWorkspace(cmd, r.ConfigHolder.Config, r.ConfigHolder.DirectoryConfig)
+	if err != nil {
+		return err
+	}
+	r.Workspace = workspace
+
+	format, err := cli.RequireOutput(cmd)
+	if err != nil {
+		return err
+	}
+	r.Format = format
+
+	accessKeyID, err := cmd.Flags().GetString("access-key-id")
+	if err != nil {
+		return err
+	}
+	secretAccessKey, err := cmd.Flags().GetString("secret-access-key")
+	if err != nil {
+		return err
+	}
+	r.AccessKeyID = accessKeyID
+	r.SecretAccessKey = secretAccessKey
+
+	if r.AccessKeyID == "" {
+		return clierrors.Message("Access Key id %q cannot be empty.", r.AccessKeyID)
+	}
+	if r.SecretAccessKey == "" {
+		return clierrors.Message("Secret Access Key %q cannot be empty.", r.SecretAccessKey)
+	}
+
+	kubeContext, ok := r.Workspace.KubernetesContext()
+	if !ok {
+		return clierrors.Message("A Kubernetes connection is required.")
+	}
+	r.KubeContext = kubeContext
+	return nil
+}
+
+// Run runs the `rad credential register aws access-key` command.
+//
+
+// Run() registers an AWS credential with the given context and workspace, and returns an error if unsuccessful.
+func (r *Runner) Run(ctx context.Context) error {
+
+	r.Output.LogInfo("Registering credential for %q cloud provider in Radius installation %q...", "aws", r.Workspace.FmtConnection())
+	client, err := r.ConnectionFactory.CreateCredentialManagementClient(ctx, *r.Workspace)
+	if err != nil {
+		return err
+	}
+	credential := ucp.AwsCredentialResource{
+		Location: to.Ptr(v1.LocationGlobal),
+		Type:     to.Ptr(cli_credential.AWSCredential),
+		Properties: &ucp.AwsAccessKeyCredentialProperties{
+			Storage: &ucp.CredentialStorageProperties{
+				Kind: to.Ptr(ucp.CredentialStorageKindInternal),
+			},
+			AccessKeyID:     &r.AccessKeyID,
+			SecretAccessKey: &r.SecretAccessKey,
+		},
+	}
+
+	err = client.PutAWS(ctx, credential)
```

Comment:
if we have access key credential configured, and run rad credential register, then we would just overwrite it, since we store exactly one credential which is "default" correct?